src/panels/design/list-shmetro.tsx

```diff
@@ -27,30 +41,110 @@ const DesignListShmetro = () => {
+                    </ListItemSecondaryAction>
+                </ListItem>
+                <Collapse in={station_number !== false} unmountOnExit>
+                    <List component="div" disablePadding className={classes.nestedList}>
+                        <ListItem>
+                            <ListItemText primary={t('design.stationNumberRailmap')} />
+                            <ListItemSecondaryAction>
+                                <Switch
+                                    color="primary"
+                                    checked={station_number_railmap !== false}
+                                    onChange={handleSwitchSecondary}
+                                />
+                            </ListItemSecondaryAction>
+                        </ListItem>
+                        <ListItem>
+                            <TextField
+                                placeholder={t('design.lineNum')}
+                                defaultValue={line_number}
+                                onChange={handleChange}
+                            />
+                        </ListItem>
+                    </List>
+                </Collapse>
+            </>
+        );
+    }, [line_number, station_number, station_number_railmap]);
+};
+
 const PanelTypeLi = () => {
     const { t } = useTranslation();
     const dispatch = useAppDispatch();
 
     const panelType = useAppSelector(store => store.param.info_panel_type);
-
     return useMemo(() => {
         const handleChange = ({ target: { value } }: ChangeEvent<{ name?: string; value: unknown }>) => {
             dispatch(setPanelType(value as PanelTypeShmetro));
         };
         return (
-            <ListItem>
-                <ListItemIcon>
-                    <Icon style={{ transform: 'rotate(180deg)' }}>credit_card</Icon>
-                </ListItemIcon>
-                <ListItemText primary={t('design.panelType.button')} />
-                <Select native value={panelType} onChange={handleChange} style={{ width: 166 }}>
-                    {Object.values(PanelTypeShmetro).map(type => (
-                        <option key={type} value={type}>
-                            {t('design.panelType.' + type)}
-                        </option>
-                    ))}
-                </Select>
-            </ListItem>
+            <>
+                <StationNumberSHMetroLi />
+                <ListItem>
+                    <ListItemIcon>
+                        <Icon style={{ transform: 'rotate(180deg)' }}>credit_card</Icon>
+                    </ListItemIcon>
+                    <ListItemText primary={t('design.panelType.button')} />
+                    <Select native value={panelType} onChange={handleChange} style={{ width: 166 }}>
+                        {Object.values(PanelTypeShmetro).map(type => (
+                            <option key={type} value={type}>
```

Comment:
Oh! Sorry my test template is using sh2020.

Reply:
Just find a problem: Shanghai Line 10 removed all station number on railmap in the 2012 style (which is the current style (without platform number version) that RMG uses) update although the station number still remain on Running-in):
![Screen Shot 2022-05-23 at 4 26 41 PM](https://user-images.githubusercontent.com/98044618/169900521-304c3f22-9183-4f94-b14c-6da3f03801f9.png)

Which now looks like this (the 2020 style also used the normal railmap without station number) (source: [https://zhuanlan.zhihu.com/p/57368457](url))
![v2-2051eed404e47fd73ca2d14b0dadc1f8_720w](https://user-images.githubusercontent.com/98044618/169900205-34b9b68f-3c3e-464f-b7c1-a492111f3521.jpg)

Reply:
@52PD  Yes, this is the reason why station number in `shmetro` has a lower priority. And since this PR is targeted for V3, we may need a rewrite for this part.

Reply:
I am going to rewrite it for v5. Thinks that we can make `stationNumberRailmap` option hidden and we only can edit it in json.

Reply:
I'm still insisting that `stationNumberRailmap` is not needed. Users can hide the station number if `showStationNumber` is set. They can download one canvas, switch the control, and download another canvas.